### PR TITLE
JSON Serializer & BSM CoreData/Part II Data Fixes

### PR DESCRIPTION
**Pull request by drewjj**
# PR Details
## Description

This pull request contains two existing feature updates:
- JSON schema output fix for duplicated 'payload.data' fields caused by the recent update of the gson library usage to be switched to jackson. This fix results in the JSON output that is put on the 'topic.Ode<message-type>Json' to be how it was prior to the recent JSON serialization library updates. This means the 'payload.data' field is no longer duplicated as 'payload.bsm', as an example for the BSM message type. This update has been applied to all of the affected ODE supported message types.  (BSM, SPaT, MAP, SRM, SSM)
- Updates to the BSM POJO builders to properly support many of the coreData and part II data fields that previously were not being supported correctly.

## Related Issue

#466 

## Motivation and Context

- The JSON serializer fix is needed because there is redundant data in the output of the latest production release of the ODE. Schemas built around how the output JSON messages were before the last production release will fail with the duplicated data.
- BSM JSON output does not contain or properly support a lot of the message fields that exist in CoreData and Part II. CDOT requires fields like SupplementaryVehicleData to contain real data instead of just being empty.

## How Has This Been Tested?

Unit tests and manual decoding of BSM, SPaT, MAP, SRM and SSM messages.

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
[ODE Contributing Guide](https://github.com/usdot-jpo-ode/jpo-ode/blob/bugfix/Pull_request_template/docs/contributing_guide.md) 
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/VehicleSizeBuilder.java [hunk 1]
```diff
@@ -48,14 +48,14 @@ public static J2735VehicleSize genericVehicleSize(JsonNode vehicleSize) {
         if (length != 0) {
             genericVehicleSize.setLength(length);
         } else {
-            genericVehicleSize.setLength(null);
+            genericVehicleSize.setLength(0);
```
Comment (dan-du-car) [inline jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/VehicleSizeBuilder.java:51]: If length == 0, the else statement is executed. Do we still need this if else statement at all?

Comment (drewjj) [reply jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/VehicleSizeBuilder.java:51]: this is a solid point

Comment (drewjj) [reply jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/VehicleSizeBuilder.java]: Addressed

File: jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/VehicleSizeBuilder.java [hunk 2]
```diff
@@ -48,14 +48,14 @@ public static J2735VehicleSize genericVehicleSize(JsonNode vehicleSize) {
...
         }
         
         int width = vehicleSize.get("width").asInt();
         if (width != 0) {
             genericVehicleSize.setWidth(width);
         } else {
-            genericVehicleSize.setWidth(null);
+            genericVehicleSize.setWidth(0);
```
[the remark this answers is not in the document]
Comment (drewjj) [reply jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/VehicleSizeBuilder.java]: Addressed

File: jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/BrakeSystemStatusBuilder.java [hunk 3]
```diff
@@ -29,8 +31,17 @@ private BrakeSystemStatusBuilder() {
    public static J2735BrakeSystemStatus genericBrakeSystemStatus(JsonNode brakesStatus) {
       J2735BrakeSystemStatus genericBrakesStatus = new J2735BrakeSystemStatus();
 
-      genericBrakesStatus.setWheelBrakes(BitStringBuilder.genericBitString(brakesStatus.get("wheelBrakes"),
+      JsonNode wheelBrakes = brakesStatus.get("wheelBrakes");
+      // Check if wheelBrakes was not set (00000), if it wasn't set it to unavailable
+      if (wheelBrakes.asText().trim().equals("00000")) {
```
Comment (dan-du-car) [inline jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/BrakeSystemStatusBuilder.java:36]: Is it possible to receive 00000 for wheelbrakes from a vehicle? https://github.com/usdot-fhwa-stol/carma-messenger/blob/f2bb8f14fd4884f619598b7739136fe81dbe2f7b/carma-messenger-core/cpp_message/src/BSM_Message.cpp#L192

Comment (dan-du-car) [reply jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/BrakeSystemStatusBuilder.java:36]: or any other values other than "10000", "01000", "00100", "00010", "00001"?

Comment (drewjj) [reply jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/BrakeSystemStatusBuilder.java:36]: Turns out it is! We are seeing it in the BSMs we are receiving in our CDOT deployment of the ODE.

Comment (drewjj) [reply jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/BrakeSystemStatusBuilder.java:36]: We get many 00000s

Comment (dan-du-car) [reply jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/BrakeSystemStatusBuilder.java:36]: good to know, thanks!

**Review by dan-du-car — APPROVED** [review]

**Pull request merged**